pkg/settings/setting.go

```diff
@@ -143,19 +143,36 @@ func NewSetting(name, def string) Setting {
 	return s
 }
 
+func GetEnvKey(key string) string {
+	return "CATTLE_" + strings.ToUpper(strings.Replace(key, "-", "_", -1))
+}
+
 func getMetadataConfig() string {
+	key := GetEnvKey("server-version")
+	rancherVersion := os.Getenv(key)
+	branch := "dev"
+	if releaseServerVersion(rancherVersion) {
+		branch = "master"
+	}
 	data := map[string]interface{}{
 		"url":                      "https://github.com/rancher/kontainer-driver-metadata.git",
-		"branch":                   "master",
+		"branch":                   branch,
 		"refresh-interval-minutes": "1440",
 	}
 	ans, err := json.Marshal(data)
-	if err == nil {
-		return string(ans)
+	if err != nil {
+		return ""
```

Comment:
do you think we should log an error? So it doesn't get lost

Reply:
updated to log error